### PR TITLE
tests: udpate the github workflow to run tests suggested by spread-filter tool

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Through this change we will run the tests suggested by the spread-filter tool following the rules located in tests/lib/spread/rules which defined for each system.

For nested tests, this ruleset will make sure that: . Always that the nested.sh lib or the assertions dir are updated, all the nested test are executed
 . When an specific nested test is added/updated, then this is executed
. If the "Run Nested" label is added to the PR, all the nested tests will be executed as well.
 . In master branch, all nested test are executed
 . In other scenarios, nested tests are not executed.

Also, in master branch, all the tests are executed (the tests defined in the matrix for each system).

### CODE REVIEW

**Comment by pedronis** [comment]
> When an specific nested test is updated, then this is executed

does that also include added?

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -810,20 +814,24 @@ jobs:
...
-                  done
-                  CHANGES_PARAM=""
-                  for CHANGE in $CHANGED_FILES; do
-                    CHANGES_PARAM="$CHANGES_PARAM -c $CHANGE"
-                  done
-                  SUGGESTED_TESTS="$SUGGESTED_TESTS $(./tests/lib/external/snapd-testing-tools/utils/spread-filter -r ./tests/lib/spread/rules/${{ matrix.rules }}.yaml -p "${{ matrix.backend }}:$SYSTEM" $CHANGES_PARAM)"
+                  # Configure parameters to run tests based on current changes
+                  # The tests are just filtered when the change is a PR
+                  # When 'Run Nested' label is added in a pr, all the nested tests have to be executed
+                  if [ -z "${{ github.event.number }}" ] || "$RUN_NESTED" = 'true'; then
```
Comment (zyga) [inline .github/workflows/test.yaml:825]: Did you mean 
```suggestion
                  if [ -z "${{ github.event.number }}" ] || [ "$RUN_NESTED" = 'true' ]; then
```

**Review by zyga — COMMENTED** [review]
One question about shell syntax.

**Comment by sergiocazzolato** [comment]
@pedronis, yes, any change "add/update" is considered

Already updated the description.

**Review by zyga — APPROVED** [review]
This looks fine to me. Thanks!

**Review by maykathm — APPROVED** [review]
thanks

**Review by bboozzoo — APPROVED** [review]
LGTM

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -818,20 +822,24 @@ jobs:
...
-                    RUN_TESTS="$RUN_TESTS ${{ matrix.backend }}:$SYSTEM:$TESTS"
-                  done
-                  CHANGES_PARAM=""
-                  for CHANGE in $CHANGED_FILES; do
-                    CHANGES_PARAM="$CHANGES_PARAM -c $CHANGE"
-                  done
-                  SUGGESTED_TESTS="$SUGGESTED_TESTS $(./tests/lib/external/snapd-testing-tools/utils/spread-filter -r ./tests/lib/spread/rules/${{ matrix.rules }}.yaml -p "${{ matrix.backend }}:$SYSTEM" $CHANGES_PARAM)"
+                  # Configure parameters to run tests based on current changes
+                  # The tests are just filtered when the change is a PR
+                  # When 'Run Nested' label is added in a pr, all the nested tests have to be executed
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply .github/workflows/test.yaml:832]: updated

**Pull request merged**